python/cuml/dask/datasets/regression.py

```diff
@@ -40,9 +45,173 @@ def create_rs_generator(random_state):
+                                chunksizes[idx], n_features,
+                                dtype, workers=[chunks_workers[idx]],
+                                pure=False)
+                  for idx, chunk in enumerate(chunksizes)]
+
+    data_dela = [da.from_delayed(dask.delayed(chunk),
+                                 shape=(chunksizes[idx], n_features),
+                                 meta=cp.zeros((1)), dtype=dtype)
+                 for idx, chunk in enumerate(data_parts)]
+    return da.concatenate(data_dela, axis=0)
+
+
+def _f_order_shuffle(X, y, n_samples, seed, features_indices):
+    local_rs = cp.random.RandomState(seed=seed)
+    samples_indices = local_rs.permutation(n_samples)
+
+    X[...] = X[samples_indices, :]
+    X[...] = X[:, features_indices]
+
+    y[...] = y[samples_indices, :]
+    return X, y
+
+
+def f_order_shuffle(client, rs, X, y, chunksizes, n_features,
+                    features_indices, n_targets, dtype):
+    X_parts = client.sync(_extract_partitions, X)
+    y_parts = client.sync(_extract_partitions, y)
+
+    chunk_seeds = rs.permutation(len(chunksizes))
+
+    shuffled = [client.submit(_f_order_shuffle, X_part, y_parts[idx][1],
+                              chunksizes[idx],
+                              chunk_seeds[idx], features_indices,
+                              workers=[w], pure=False)
+                for idx, (w, X_part) in enumerate(X_parts)]
+
+    X_shuffled = [client.submit(get_X, f, pure=False)
+                  for idx, f in enumerate(shuffled)]
+    y_shuffled = [client.submit(get_labels, f, pure=False)
+                  for idx, f in enumerate(shuffled)]
+
+    X_dela = [da.from_delayed(dask.delayed(Xs),
+                              shape=(chunksizes[idx], n_features),
+                              meta=cp.zeros((1)),
+                              dtype=dtype)
+              for idx, Xs in enumerate(X_shuffled)]
+
+    y_dela = [da.from_delayed(dask.delayed(ys),
+                              shape=(chunksizes[idx], n_targets),
+                              meta=cp.zeros((1)),
+                              dtype=dtype)
+              for idx, ys in enumerate(y_shuffled)]
+
+    return da.concatenate(X_dela, axis=0), da.concatenate(y_dela, axis=0)
+
+
+def convert_C_to_F_order(client, X, chunksizes, n_features, dtype):
+    X_parts = client.sync(_extract_partitions, X)
+    X_converted = [client.submit(cp.array, X_part, copy=False, order='F',
+                                 workers=[w])
+                   for idx, (w, X_part) in enumerate(X_parts)]
+
+    X_dela = [da.from_delayed(dask.delayed(Xc),
+                              shape=(chunksizes[idx], n_features),
+                              meta=cp.zeros((1)),
+                              dtype=dtype)
+              for idx, Xc in enumerate(X_converted)]
+    return da.concatenate(X_dela, axis=0)
+
+
+def generate_chunks_for_qr(total_size, min_size, n_parts):
```

Comment:
If these functions are meant to be internally used, we should prefix them with `_`. If they are meant to be externally used, we should add pydocs.